test/src/main/java/org/axonframework/test/aggregate/AggregateTestFixture.java

```diff
@@ -178,6 +166,16 @@ public AggregateTestFixture(Class<T> aggregateType, boolean useStateStorage) {
         return this;
     }
 
+    @Override
+    public FixtureConfiguration<T> useStateStorage() {
+        return this.registerRepository(new InMemoryRepository<>(aggregateType,
```

Comment:
Should this method overwrite an existing repository, or rather throw an exception?

Reply:
Oh, my apologies, my explanation towards this option wasn't clear enough I think.
Using the `boolean` as you did fit the construction of an `InMemoryRepository` quite nicely inside the `ensureRepositoryConfiguration()` method.

So what my intent was with this method, is that a `boolean` is being set instead of the construction of the `Repository`.
That reverts back some of the changes you made in your previous PR as far as I recall.

Reply:
Sorry for my misunderstanding, I changed the PR so that `ensureRepositoryConfiguration()` is now responsible for creating the repository.

I also added a check in the `givenState()` method, so u`seStateStorage()` will be called, if no repository is configured. 
I think this is necessary to prevent the change from breaking existing code.

Could this be a problem if someone is using `givenState()` while using event sourcing?